demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -201,14 +196,11 @@ int main(int argc, char *argv[]) {
             throw std::logic_error("Class labels are not supported with IR version older than 10");
         }
 
-        if (static_cast<int>(labels.size()) != num_classes) {
+        if (!labels.empty() && static_cast<int>(labels.size()) != num_classes) {
```

Comment:
Why is `!labels.empty()` needed? It should be fine without it

Reply:
We have three different cases here:
1. Labels aren't set at all (`-labels` omitted)
2. Labels have no `background` class
3. Labels number differs from number of classes.

First case we simply ignore. In second case we add `fake` label. Third case is an error. 
If we leave this section:
```
        if (static_cast<int>(labels.size()) != num_classes) {
            if (static_cast<int>(labels.size()) == (num_classes - 1))  
                labels.insert(labels.begin(), "fake");
            else {
                throw std::runtime_error("The number of labels is different from numbers of model classes");
            }                
        }
```
it will break everytime without `-labels` option specified.

Reply:
You are right, ok.